app/lib/task/cloudcompute/fakecloudcompute.dart

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:clock/clock.dart';
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart';
+
+import 'cloudcompute.dart';
+
+final _log = Logger('pub.fakecloudcompute');
+
+@sealed
+class FakeCloudCompute extends CloudCompute {
+  var _nextInstanceId = 1;
+  final _instances = <FakeCloudInstance>{};
+
+  @override
+  final List<String> zones = List.unmodifiable(['zone-a', 'zone-b']);
+
+  @override
+  String generateInstanceName() => 'instance-${_nextInstanceId++}';
+
+  @override
+  Future<FakeCloudInstance> createInstance({
+    required String zone,
+    required String name,
+    required String dockerImage,
+    required List<String> arguments,
+    required String description,
+  }) async {
+    if (!zones.contains(zone)) {
+      throw ArgumentError.value(zone, 'zone', 'must be a valid zone');
+    }
+    // Enforce same limits as GCE
+    if (arguments.fold<int>(0, (a, b) => a + b.length) > 100 * 1024) {
+      throw ArgumentError.value(
+        arguments,
+        'arguments',
+        'must be less than 100KiB',
+      );
+    }
+    if (_instances.any((i) => i.name == name)) {
+      throw StateError('instance "$name" have already been used once!');
+    }
+
+    final instance = FakeCloudInstance._(
+      name: name,
+      zone: zone,
+      created: clock.now().toUtc(),
+      state: InstanceState.pending,
+      dockerImage: dockerImage,
+      arguments: arguments,
+      description: description,
+    );
+    _log.info('Creating instance "$name"');
+    _instances.add(instance);
+    return instance;
+  }
+
+  @override
+  Stream<FakeCloudInstance> listInstances() => Stream.fromIterable(_instances);
+
+  @override
+  Future<void> delete(String zone, String name) async {
+    if (!_instances.any((i) => i.name == name && i.zone == zone)) {
+      // This is not really a problem, GoogleCloudCompute should might throw
+      // some exception about the instance not being found. Exact behavior of
+      // the API is not specified.
+      throw Exception('instance "$name" does not exist');
+    }
+
+    // Let's make the operation take a minute, and then remove the instance!
+    _log.info('Deleting instance "$name"');
+    await Future.delayed(Duration(minutes: 1));
+    _instances.removeWhere((i) => i.name == name && i.zone == zone);
+  }
+
+  /// Change state of instance with [name] to [InstanceState.running].
+  void fakeStartInstance(String name) {
+    if (!_instances.any((i) => i.name == name)) {
+      throw StateError('instance "$name" does not exist');
+    }
+
+    final instance = _instances.firstWhere((i) => i.name == name);
```

Comment:
```suggestion
    final instance = _instances.firstWhereOrNull((i) => i.name == name);
    if (instance == null) {
      throw StateError('instance "$name" does not exist');
    }
```
(requires package:collection)